pkg/commands/branches.go

```diff
@@ -101,6 +101,39 @@ func (c *GitCommand) GetBranchUpstreamDifferenceCount(branchName string) (string
 	return c.GetCommitDifferences(branchName, branchName+"@{u}")
 }
 
+// TryGetMainBranch tries to get the main branch
+// For old repo's this is usually master and for newer once it is main
+func (c *GitCommand) TryGetMainBranch() string {
+	// Firstly try to get the main branch from the origins
+	cmdStr := `git branch --remotes --list '*/HEAD' --format '%(symref:short)'`
+	output, _ := c.OSCommand.RunCommandWithOutput(cmdStr)
+	lines := strings.Split(strings.TrimSpace(output), "\n")
+	for _, line := range lines {
+		parts := strings.SplitN(line, "/", 2)
+		if len(parts) == 2 {
+			return parts[len(parts)-1]
+		}
+	}
+
+	// If we didn't find a main branch from the remotes we check if there is a main or master branch
+	cmdStr = `git branch --format "%(refname:short)" --list "master" --list "main"`
+	output, _ = c.OSCommand.RunCommandWithOutput(cmdStr)
+	lines = append(lines[:0], strings.Split(strings.TrimSpace(output), "\n")...)
+	for _, line := range lines {
+		if line == "master" || line == "main" {
```

Comment:
I think we should first see if any of them are master and then if that fails, check if any are main. Because I've seen a couple repos that just happen to have a 'main' branch but actually use 'master'. If the repo intentionally uses a 'main' branch, we should expect them to have removed the 'master' branch.

Reply:
Fixed